api/server.go

```diff
@@ -45,7 +48,9 @@ func (s *Server) routes() http.Handler {
 	mux := http.NewServeMux()
 
 	mux.HandleFunc("GET /", s.HelloServerHandler)
-	mux.HandleFunc("GET /stocks", s.GetStockBySymbolHandler)
+	stockHandler := &StockHandler{store: s.store, apiKey: s.apiKey}
+	mux.HandleFunc("GET /stocks", stockHandler.GetStockBySymbolHandler)
```

Comment:
Why create a whole new struct for this? The `Server` already has access to the store and apiKey, so I don't understand why create a separate struct

Reply:
It can still be in separate files though 👍